src/RulesEngineEditor/Pages/RulesEngineEditorPage.razor

```diff
@@ -166,9 +190,9 @@
                     }
                 }
             </div>
-            <div>
+            <div hidden="@ShowDesigner">
                 <span class="JSONErrors">@inputJSONErrors</span>
-                <textarea @bind="InputJSON" class="jsonTextArea" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" />
+                <textarea @bind="InputJSON" rows="@(Regex.Matches(InputJSON, "\n").Count + 1)" class="jsonTextArea" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" />
```

Comment:
`InputJSON.ToCharArray().Count(c => c == '\n') + 1` ?

Reply:
Text area needs rows number to expand in full height, otherwise it is shrinked to minimum. Json strings contains "\n" for each new row + 1 for the first line :). It seems to work probably not the smartest solution

Reply:
sure but why regex and not just count the new lines